packages/openai/src/chat.ts

```diff
@@ -279,10 +279,14 @@ export class OpenAIChatModel implements IChatModel<ChatCompletionCreateParams, O
       };
 
       if (message.tool_calls && message.tool_calls.length > 0) {
-        return this.send(modelMessage, {
-          ...options,
-          messages: memory,
-        });
+        if (!options.disableAutomaticFunctionCalling) {
```

Comment:
personal pref but would it be less confusing if called it enableAutoFunctionCalling - the double negation took me a bit to digest

Reply:
Plus 1 to this! :)

Reply:
nit: might be simpler to refactor like: 
```typescript
if (message.tool_calls && message.tool_calls.length > 0) {
  // Skip function calling if explicitly disabled
  if (!options.enableFunctionCalling) {
    this._log.debug(`Function calling is disabled; skipping function call (total calls: ${message.tool_calls.length})`);
    return modelMessage;
  }
  
  return this.send(modelMessage, {
    ...options,
    messages: memory,
  });
```

Reply:
Cool with this -- I usually agree with this, but in places where the default is _true_ AND the input is optional, i tend to prefer disable.

Corina your code will trigger if enableFunctionCall == null though.

Reply:
(updated to `!!options.enableFunctionCalling` and keeping it the way it is)